src/Containers/Microsoft.NET.Build.Containers/LocalDaemons/DockerCli.cs

```diff
@@ -275,18 +275,44 @@ private static async Task WriteImageToStreamAsync(BuiltImage image, ImageReferen
         // Try to find the docker or podman cli.
         // On systems with podman it's not uncommon for docker to be an alias to podman.
         // We try to find podman first so we can identify those systems to be using podman.
-        foreach (var command in new[] { PodmanCommand, DockerCommand })
+        var podmanCommand = TryRunVersionCommandAsync(PodmanCommand, cancellationToken);
+        var dockerCommand = TryRunVersionCommandAsync(DockerCommand, cancellationToken);
+
+        await Task.WhenAll(
+            podmanCommand,
+            dockerCommand
+        ).ConfigureAwait(false);
+
+        // be explicit with this check so that we don't do the link target check unless it might actually be a solution.
+        if (dockerCommand.Result && podmanCommand.Result && IsPodmanAlias())
         {
-            if (await TryRunVersionCommandAsync(command, cancellationToken))
-            {
-                _commandPath = command;
-                break;
-            }
+            _commandPath = PodmanCommand;
+        }
+        else if (dockerCommand.Result)
+        {
+            _commandPath = DockerCommand;
+        }
+        else if (podmanCommand.Result)
+        {
+            _commandPath = PodmanCommand;
         }
 
         return _commandPath;
     }
 
+    private static bool IsPodmanAlias()
+    {
+        var dockerPath = FindFullPathFromPath(DockerCommand);
+        var podmanPath = FindFullPathFromPath(PodmanCommand);
+        if (dockerPath is null || podmanPath is null)
+        {
+            return false;
+        }
+
+        var fi = new FileInfo(dockerPath);
+        return fi.LinkTarget == podmanPath;
```

Comment:
`docker` isn't a link on Fedora, it's a script:
```sh
$ cat /usr/bin/docker 
#!/usr/bin/sh
[ -e /etc/containers/nodocker ] || \
echo "Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg." >&2
exec /usr/bin/podman "$@"
```
We may use something from the following, like the presence of `buildahVersion`.
```json
$ docker info -f '{{ json }}'
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
{
  "host": {
    "arch": "amd64",
    "buildahVersion": "1.31.2",
    "cgroupManager": "systemd",
    "cgroupVersion": "v2",
    "cgroupControllers": [
      "cpu",
      "io",
      "memory",
      "pids"
    ],
    "conmon": {
      "package": "conmon-2.1.7-2.fc37.x86_64",
      "path": "/usr/bin/conmon",
      "version": "conmon version 2.1.7, commit: "
    },
    "cpus": 12,
    "cpuUtilization": {
      "userPercent": 8.06,
      "systemPercent": 3.04,
      "idlePercent": 88.9
    },
    "databaseBackend": "boltdb",
    "distribution": {
      "distribution": "fedora",
      "variant": "workstation",
      "version": "37"
    },
    "eventLogger": "journald",
    "freeLocks": 1843,
    "hostname": "fedora",
    "idMappings": {
      "gidmap": [
        {
          "container_id": 0,
          "host_id": 1000,
          "size": 1
        },
        {
          "container_id": 1,
          "host_id": 100000,
          "size": 65536
        }
      ],
      "uidmap": [
        {
          "container_id": 0,
          "host_id": 1000,
          "size": 1
        },
        {
          "container_id": 1,
          "host_id": 100000,
          "size": 65536
        }
      ]
    },
    "kernel": "6.4.7-100.fc37.x86_64",
    "logDriver": "journald",
    "memFree": 23019454464,
    "memTotal": 33385041920,
    "networkBackend": "netavark",
    "networkBackendInfo": {
      "backend": "netavark",
      "version": "netavark 1.7.0",
      "package": "netavark-1.7.0-1.fc37.x86_64",
      "path": "/usr/libexec/podman/netavark",
      "dns": {
        "version": "aardvark-dns 1.7.0",
        "package": "aardvark-dns-1.7.0-1.fc37.x86_64",
        "path": "/usr/libexec/podman/aardvark-dns"
      }
    },
    "ociRuntime": {
      "name": "crun",
      "package": "crun-1.8.5-1.fc37.x86_64",
      "path": "/usr/bin/crun",
      "version": "crun version 1.8.5\ncommit: b6f80f766c9a89eb7b1440c0a70ab287434b17ed\nrundir: /run/user/1000/crun\nspec: 1.0.0\n+SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL"
    },
    "os": "linux",
    "remoteSocket": {
      "path": "/run/user/1000/podman/podman.sock"
    },
    "serviceIsRemote": false,
    "security": {
      "apparmorEnabled": false,
      "capabilities": "CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT",
      "rootless": true,
      "seccompEnabled": true,
      "seccompProfilePath": "/usr/share/containers/seccomp.json",
      "selinuxEnabled": true
    },
    "slirp4netns": {
      "executable": "/usr/bin/slirp4netns",
      "package": "slirp4netns-1.2.0-8.fc37.x86_64",
      "version": "slirp4netns version 1.2.0\ncommit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383\nlibslirp: 4.7.0\nSLIRP_CONFIG_VERSION_MAX: 4\nlibseccomp: 2.5.3"
    },
    "pasta": {
      "executable": "/usr/bin/pasta",
      "package": "passt-0^20230625.g32660ce-1.fc37.x86_64",
      "version": "pasta 0^20230625.g32660ce-1.fc37.x86_64\nCopyright Red Hat\nGNU Affero GPL version 3 or later \u003chttps://www.gnu.org/licenses/agpl-3.0.html\u003e\nThis is free software: you are free to change and redistribute it.\nThere is NO WARRANTY, to the extent permitted by law.\n"
    },
    "swapFree": 8589930496,
    "swapTotal": 8589930496,
    "uptime": "0h 4m 37.00s",
    "linkmode": "dynamic"
  },
  "store": {
    "configFile": "/home/tmds/.config/containers/storage.conf",
    "containerStore": {
      "number": 34,
      "paused": 0,
      "running": 2,
      "stopped": 32
    },
    "graphDriverName": "overlay",
    "graphOptions": {
      
    },
    "graphRoot": "/home/tmds/.local/share/containers/storage",
    "graphRootAllocated": 510405902336,
    "graphRootUsed": 193626595328,
    "graphStatus": {
      "Backing Filesystem": "btrfs",
      "Native Overlay Diff": "true",
      "Supports d_type": "true",
      "Using metacopy": "false"
    },
    "imageCopyTmpDir": "/var/tmp",
    "imageStore": {
      "number": 192
    },
    "runRoot": "/run/user/1000/containers",
    "volumePath": "/home/tmds/.local/share/containers/storage/volumes",
    "transientStore": false
  },
  "registries": {
    "search": [
  "registry.fedoraproject.org",
  "registry.access.redhat.com",
  "docker.io",
  "quay.io"
]
  },
  "plugins": {
    "volume": [
      "local"
    ],
    "network": [
      "bridge",
      "macvlan",
      "ipvlan"
    ],
    "log": [
      "k8s-file",
      "none",
      "passthrough",
      "journald"
    ],
    "authorization": null
  },
  "version": {
    "APIVersion": "4.6.1",
    "Version": "4.6.1",
    "GoVersion": "go1.19.10",
    "GitCommit": "",
    "BuiltTime": "Fri Aug 11 00:07:55 2023",
    "Built": 1691705275,
    "OsArch": "linux/amd64",
    "Os": "linux"
  }
}
```

Reply:
That's reasonable to do - I wonder if expanding `GetConfig()` for a Podman use case would be worthwhile as well...

Reply:
Thank you for the check! I should have done more research before starting - I think your idea is a good one and will go that route in the next commits here, likely tomorrow.

Reply:
Ok, I've finally implemented this. Can you give that a try and see how it looks on your systems? I'll also be trying with sdk-container-demo (via the package) when I get the chance.

Reply:
Thanks for the testing and the PR, I've merged it and we'll let CI take a look at the results.

Reply:
Could this still be in rc1?

Reply:
Unfortunately not - we're past the deadlines for that at this point.